lib/pagseguro/subscription/request_serializer.rb

```diff
@@ -0,0 +1,124 @@
+module PagSeguro
+  class Subscription
+    class RequestSerializer
+      # Get the object
+      attr_reader :object
+
+      def initialize(object)
+        @object = object
+      end
+
+      def serialize
+        Nokogiri::XML::Builder.new(encoding: PagSeguro.encoding) do |xml|
+          xml.send(:directPreApproval) {
+            xml.send(:plan, object.plan) if object.plan
+            xml.send(:reference, object.reference) if object.reference
+            serialize_sender(xml, object.sender)
+            serialize_payment_method(xml, object.payment_method)
+          }
+        end.to_xml(save_with:
```

Comment:
I did not like this way :/ … I prefer to split into other two or use `}`.

#justsaying